### ISSUE TITLE
Permissions Possibly Unsafe?

### ISSUE THREAD

**Issue by ghost** [paperbenni/bash#9]
There are 30 shell programs you have with `755` permissions, which means everyone can read and execute them. Are you sure that's what you want? Someone wouldn't, at least ordinarily, be able to do a whole lot of damage, but maybe they could do something undesired.

If these are one-off scripts intended to be used only by you and are not installed anywhere in `PATH`, then I recommend `700`, as is also recommended by RHEL.

The command I used to determine this, from within the repository's root directory:

```bash
find -not -path '*.git*' -type f -perm 755
```

**Comment by paperbenni** [comment]
The files are actually not meant to executed at all. None of them do anything besides defining functions. The entire project is meant to emulate the import statement of python. 
I'm not sure how familiar you are with python but basically what import does is it sources a script from pythonpath (in most cases managed by pip) with the given name. It also only sources a script once even when imported twice, in case a script (or module in python terminology) that's imported already contains an import statement, so dulpicate importing and recursion are avoided. 
The difference between python packages and pb packages are that they are for bash (duh), and if not found locally are fetched from the git repo. (afterwards the local copy gets used). pb then just sources the script. 
So at the top of my scripts I have a bunch of pb statements and then use the imported (=sourced) functions in the main part of the script. 
Long story short, if they are for sourcing, you can't do it on accident and none of the permissions except read really matter.

**Comment by ghost** [comment]
That makes a lot of sense. Did wonder what that pb thing was all about. lol Very cool stuff. The module thing is familiar to me because of `use` in Perl, and my super-amateur experience in Python just about lends me to remember importing.

So you've essentially got a function library but they're spread out over lots of files and intended to be sourced; did I get that right?

I geuess the permissions are irrelevant then, since things seem protected within the function.
